src/threechances.cpp

```diff
@@ -35,36 +35,38 @@ void ThreeChances::initialize(HWND hwnd) {
 	if (!playerMaleTexture.initialize(graphics, PLAYER_MALE_IMAGE))
 		throw(GameError(gameErrorNS::FATAL_ERROR, "Error initializing male texture"));
 
-	if (!map.initialize(graphics, 0, 0, 0, &mapTexture))
+	if (!duckTexture.initialize(graphics, DUCK_IMAGE))
+		throw(GameError(gameErrorNS::FATAL_ERROR, "Error initializing duck texture"));
+
+	if (!map.initialize(this, &mapTexture))
 		throw(GameError(gameErrorNS::FATAL_ERROR, "Error initializing map"));
 
-	if (!playerMale.initialize(graphics, TILE_SIZE, TILE_SIZE, 3, &playerMaleTexture))
+	if (!player.initialize(this, TILE_SIZE, TILE_SIZE, PLAYER_COLS, &playerMaleTexture, PLAYER_DATA))
 		throw(GameError(gameErrorNS::FATAL_ERROR, "Error initializing male player"));
 
-	playerMale.setScale((float)SCALE);
-	playerMale.setX(TILE_SIZE * SCALE * 3);
-	playerMale.setY(TILE_SIZE * SCALE * 3);
-
-	// Set initial male position, should follow grid later
-	playerMale.setLoop(false);
-	playerMale.setFrames(PLAYER_START_FRAME, PLAYER_END_FRAME);	// animation frames
-	playerMale.setFrameDelay(PLAYER_ANIMATION_DELAY);
-	playerMale.setCurrentFrame(3);
+	if (!duck.initialize(this, TILE_SIZE, TILE_SIZE, DUCK_COLS, &duckTexture, DUCK_DATA))
+		throw(GameError(gameErrorNS::FATAL_ERROR, "Error initializing duck monster"));
 
+	// Set map to default scale and starting position
 	map.setScale((float)SCALE);
 	map.setX(-(TILE_SIZE * SCALE * ((float)stage->getStartTile().x - 3)));
 	map.setY(-(TILE_SIZE * SCALE * ((float)stage->getStartTile().y - 3)));
 
-	stage->logLayout();
+	player.setScale((float)SCALE);
```

Comment:
Not sure if this will work but can you add in the Entity class' init function `spriteData.scale = (FLOAT)SCALE;`? Then we wouldnt have to set the scale everything we create a new entity